File: 24/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func isAnagram(a, b string) bool {
	if len(a) != len(b) {
		return false
	}
	for _, r := range b {
		if !strings.ContainsRune(a, r) {
			return false
		}
	}
	return true
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Println("Enter two strings and I'll tell you if then are anagrams: ")

	fmt.Print("Enter the first string: ")
	scanner.Scan()
	if scanner.Err() != nil {
		os.Exit(1)
	}
	first := scanner.Text()

	fmt.Print("Enter the second string: ")
	scanner.Scan()
	if scanner.Err() != nil {
		os.Exit(1)
	}
	second := scanner.Text()

	ok := isAnagram(first, second)

	not := ""
	if !ok {
		not = "not "
	}
	fmt.Printf("\"%s\" and \"%s\" are %sanagrams.\n", first, second, not)
}

File: 24/cpp/main.cc
#include <iostream>

#include "./anagram.h"

inline static std::string surround_dq(std::string s)
{
    return "\"" + s + "\"";
}

int main()
{
    std::string first, second;

    std::cout << "Enter two strings and I'll tell you if then are anagrams: " << std::endl;

    std::cout << "Enter the first string: ";
    std::getline(std::cin, first);

    std::cout << "Enter the second string: ";
    std::getline(std::cin, second);

    bool anagram = is_anagram(first, second);
    std::cout
        << surround_dq(first)
        << " and "
        << surround_dq(second)
        << " are "
        << (anagram ? "" : "not ")
        << "anagrams."
        << std::endl;

    return 0;
}

